hazelcast/src/test/java/com/hazelcast/cluster/ClusterClockTest.java

```diff
@@ -0,0 +1,99 @@
+package com.hazelcast.cluster;
+
+
+import com.hazelcast.config.Config;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.instance.Node;
+import com.hazelcast.instance.TestUtil;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
+public class ClusterClockTest extends HazelcastTestSupport {
+
+    @Before
+    @After
+    public void cleanup() {
+        Hazelcast.shutdownAll();
+    }
+
+    @Test
+    public void test_start_time_single_node_cluster() throws Exception {
+        Config config = new Config();
+        HazelcastInstance h1 = Hazelcast.newHazelcastInstance(config);
```

Comment:
For this case yes, I have tried with TestHazelcastInstanceFactory did not work, because handleJoinRequest is not called without network testing, and it is where I set clusterStartTime and clusterId.